### ISSUE TITLE
Instant map update after changing focus to a different document / Save settings on chekbox-click / Save vertical scroll value per document

### ISSUE THREAD

**Issue by KittenpawFromsoftfur** [oleg-shilo/PyMap.VSIX#6]
Hello oleg, first of all a very big thank you for creating the best extension ever. I have been looking for a Visual Studio outline that does not suck, for ages, not even CodeMaid could satisfy me because you cannot hide the entire list of member-variables, all I need is the functions and thats it.

So after trying out your extension I was in bliss, but I figured I could add the few changes that were missing for me and as a thank you, I am sharing it now with you.

Features added:
- Now the Map refreshes as soon as a different document is focused (No need for manual input like Ctrl+S)
- All settings are saved when a checkbox is clicked (Before that you had to do Ctrl+Mouse wheel in order to save)
- Vertical Scrollbar value of the CodeMap can be saved for each document. You can disable this with the new checkbox "Remember scroll"
- Sort option will now also be saved (I think you forgot to include it in the serialize Settings method)

Changed files:
- SyntaxParser.cs
- ToolWindow1Control.xaml
- ToolWindow1Control.xaml.cs

[PyMap.VSIX-2.0.2.0.zip](https://github.com/oleg-shilo/PyMap.VSIX/files/10336528/PyMap.VSIX-2.0.2.0.zip)

**Comment by oleg-shilo** [comment]
I am glad you found this extension being useful.

I completely share your pain with the existing "code outline" solutions. 

Being an extension developer by myself I am very respectful of other extensions authors desire to see their products evolving certain way. Thus I have done a few attempts to contribute the PRs to CodeMaid team but I did not succeed convincing them that they are not capturing all code navigation use-cases and actually overlooking the most important "code outline" user experience - navigation. I have described all this in the `readme.md`.

Thus I have created this extension for myself but shared with all. And it looks like it is getting a bit more and more attention.

As for your code contribution, ironically, I was already working on the feature that that deliverers the same user experience as you are looking for with your "save scroll position per doc". The latest release (today) allows synchronizing tree selection and scroll-to-view position with the caret position in the document. This synchronization can also be done automatically as you navigate within the file.   

![image](https://user-images.githubusercontent.com/16729806/210564008-b73d6e22-e97a-4aad-98ed-c6c96220831d.png)

Thus I did not use your implementation of this feature but did take the other two.

Thank you for your contribution.
